v2/oauth/multiple-frontends-with-a-single-backend.mdx

```diff
@@ -0,0 +1,288 @@
+
+:::
+
+</TabItem>
+
+<TabItem value="python">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Python SDK.
+
+:::
+
+</TabItem>
+
+</BackendSDKTabs>
+
+###  Configure the Authorization Service Frontend 
+
+Initialize the `OAuth2Provider` recipe on the frontend of your **Authorization Service**. 
+
+:::info
+
+If you want to use your own custom UI check our [separate guide](/docs/oauth/customizations/custom-ui) that explains
+all the steps that you have to take into account.
+
+:::
+
+<OAuthFrontendTabs>
+
+
+<TabItem value="react">
+
+Update the `SuperTokensConfig` variable to adjust to the new flow. 
+
+1. Add the import statement for the recipe.
+
+```typescript 
+import OAuth2Provider from "supertokens-auth-react/recipe/oauth2provider";
+```
+
+2. Update the list of recipes to also include the new initialization.
+
+```tsx
+OAuth2Provider.init()
+```
+
+3. Update the **Session** recipe initialization to use `header` authentication.
+
+```tsx
+Session.init({ authMode: "header" })
+```
+    
+</TabItem>
+
+
+<TabItem value="angular">
+
+1. Update the `AuthComponent` so that it also includes the `OAuth2Provider` recipe.
+You will have to add a new item in the `recipeList` array, inside the `supertokensUIInit` call.
+
+```typescript
+
+supertokensUIOAuth2Provider.init(),
+
+```
+
+2. Update the `supertokensUISession` init step to use `header` authentication.
+        
+```typescript
+supertokensUISession.init({
+    authMode: "header"
+})
+
+```
+
+3. Update the `Supertokens.init` call so that the `Session` recipe uses `header` authentication.
+
+```typescript 
+
+Session.init({ authMode: "header" })
+
+```
+
+</TabItem>
+
+<TabItem value="vue">
+
+1. Update the `AuthView` component so that it also includes the `OAuth2Provider` recipe.
+You will have to add a new item in the `recipeList` array, inside the `supertokensUIInit` call.
+
+```typescript
+
+supertokensUIOAuth2Provider.init(),
+
+```
+
+
+2. Update the `supertokensUISession` init step to use `header` authentication.
+        
+```typescript
+supertokensUISession.init({
+    authMode: "header"
+})
+
+```
+
+3. Update the `Supertokens.init` call so that the `Session` recipe uses `header` authentication.
+
+```typescript 
+
+Session.init({ authMode: "header" })
+
+```
+
+    
+</TabItem>
+    
+</OAuthFrontendTabs>
+
+
+## 4. Update the login flow in your frontend applications
+
+In your `frontend` applications you will have to add a login action that will direct the user to the authentication page.
+
+The user should first be redirected to the `backend` login endpoint.
+There the `backend` will redirect the user to the actual [**Authorization Service**](/docs/oauth/introduction#authorization-service) login UI.
+After the user has logged in they will be redirected to the `backend` callback URL.
+Then the authentication session will be created and the `backend` will send it to the user agent as a cookie.
```

Comment:
You should first mention that you need to setup supertokens session in the frontend application here, and you can do that by following https://supertokens.com/docs/session/quick-setup/frontend.

Then talk about setting it to header based auth, and say why we need to do it - cause each of these frontends will be third party domains to the backend, and third party cookies are not allowed by browsers like safari.

Then we need to talk about using an oauth lib for the framework and setting the various values in that. 

Then finally, when the oauth login is done, we can tell users to call some API (which one - need to have the curl command) which will exchange the oauth tokens for a supertokens session?

Reply:
resolved